hs/t/n/map-dset.txt

```diff
@@ -7,14 +7,27 @@ Verification failed:
   msg: "m[A] is Some(ap)"
   at ./map-dset.rsh:29:10:application
 
-  // Violation witness
-  const address_Alice = Address!val!0;
-  //    ^ from builtin at ./map-dset.rsh:4:30:application
-  const a/27 = 1;
-  //    ^ from evaluating interact("Alice")."get"() at ./map-dset.rsh:16:51:application
-
-  // Theorem formalization
-  require(digest(map0[address_Alice]) == digest(a/27));
+  // Violation Witness
+
+  const a/27 = "Alice".interact.get();
+  //    ^ could = 1
+  //      from: ./map-dset.rsh:16:51:application
+  const b/36 = "Bob".interact.get();
+  //    ^ could = 0
+  //      from: ./map-dset.rsh:21:51:application
+
+
+  // Theorem Formalization
+
+  const map0_0/57 = new Map();
+
+  const map0_1/70 = set(map0_0/57["Address!val!0"], Just a/27);
```

Comment:
I'm surprised that `map0_0` didn't get substituted.

Reply:
It didn't get subbed because `canDupe SMTMapNew` is false. I figured the only `SynthExpr` that could be dupe'd is `SMTMapRef`. Should new maps be able to be duplicated?

Reply:
Well, we're not going to be executing these, they are just informative. So I agree that if we're talking about a real program we should not turn `x = new(); y = x; z = x;` into `y = new(); z = new();` because we would lose sharing. In this case though, we are really saying `x = something-that-came-from-new; y = x; z = x;` and in that case, it is more like the pointer that comes from `new` than the call itself. Same with set.